drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,398 @@
+
+	int rc = 0;
+
+	/*
+	 * the display wraps around by itself, just write the text and update the position data
+	 */
+	for (int i = 0; i < len; ++i) {
+		uint8_t character = text[i];
+
+		/*
+		 * customer characters require a special command, so check for custom char
+		 */
+		if ((character & 0xf8) == 0x08) {
+			rc = auxdisplay_serlcd_send_command(dev, SERLCD_COMMAND_WRITE_CUSTOM_CHAR +
+									 (character & 0x07));
+			if (!rc) {
+				return rc;
+			}
+			auxdisplay_serlcd_advance_current_position(dev);
+		} else if (character == SERLCD_BEGIN_COMMAND ||
+			   character == SERLCD_BEGIN_SPECIAL_COMMAND) {
+			/*
+			 * skip these characters in text, as they have a special meaning, if
+			 * required a custom character can be used as replacement
+			 */
+			continue;
+		} else {
+			rc = i2c_write_dt(&config->bus, text, len);
+			if (!rc) {
+				return rc;
+			}
+			auxdisplay_serlcd_advance_current_position(dev);
+		}
+	}
+
+	return rc;
+}
+
+static int auxdisplay_serlcd_init(const struct device *dev)
+{
+	const struct auxdisplay_serlcd_config *config = dev->config;
+
+	if (!device_is_ready(config->bus.bus)) {
+		return -ENODEV;
+	}
+
+	auxdisplay_serlcd_clear(dev);
+
+	return 0;
+}
+
+static const struct auxdisplay_driver_api auxdisplay_serlcd_auxdisplay_api = {
+	.display_on = auxdisplay_serlcd_display_on,
+	.display_off = auxdisplay_serlcd_display_off,
+	.cursor_set_enabled = auxdisplay_serlcd_cursor_set_enabled,
+	.position_blinking_set_enabled = auxdisplay_serlcd_position_blinking_set_enabled,
+	.cursor_position_set = auxdisplay_serlcd_cursor_position_set,
+	.cursor_position_get = auxdisplay_serlcd_cursor_position_get,
+	.capabilities_get = auxdisplay_serlcd_capabilities_get,
+	.clear = auxdisplay_serlcd_clear,
+	.custom_character_set = auxdisplay_serlcd_custom_character_set,
+	.write = auxdisplay_serlcd_write,
+};
+
+#define AUXDISPLAY_SERLCD_INST(inst)                                                               \
+	static const struct auxdisplay_serlcd_config auxdisplay_serlcd_config_##inst = {           \
+		.capabilities =                                                                    \
+			{                                                                          \
+				.columns = DT_INST_PROP(inst, columns),                            \
+				.rows = DT_INST_PROP(inst, rows),                                  \
+				.mode = 0,                                                         \
+				.brightness.minimum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,              \
+				.brightness.maximum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,              \
+				.backlight.minimum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,               \
+				.backlight.maximum = AUXDISPLAY_LIGHT_NOT_SUPPORTED,               \
+				.custom_characters = 8,                                            \
+				.custom_character_width = 5,                                       \
+				.custom_character_height = 8,                                      \
```

Comment:
these could use a define (they show up in other spots anyway)

Reply:
Done